src/main/java/mate/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,35 @@
+package mate.web.filter;
+
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse, FilterChain filterChain)
+                throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+        HttpSession session = request.getSession();
+
+        Long id = (Long) session.getAttribute("id");
+        if (id == null && (request.getServletPath().equals("/drivers/login")
+                || request.getServletPath().equals("/drivers/add"))) {
```

Comment:
use init() and Set to save all allowed urls plus include "/index" and "/" as they just display your main menu. But on click you will be redirected to login page